ush/python/pygfs/task/snowens_analysis.py

```diff
@@ -2,429 +2,435 @@
                     logit,
                     Executable,
                     WorkflowException)
-from pygfs.task.analysis import Analysis
+from pygfs.jedi import Jedi
 
 logger = getLogger(__name__.split('.')[-1])
 
 
-class SnowEnsAnalysis(Analysis):
+class SnowEnsAnalysis(Task):
     """
-    Class for global ensemble snow analysis tasks
+    Class for JEDI-based global snow ensemble analysis tasks
     """
 
     @logit(logger, name="SnowEnsAnalysis")
-    def __init__(self, config):
+    def __init__(self, config: Dict[str, Any]):
+        """Constructor global snow ensemble analysis task
+
+        This method will construct a global snow ensemble analysis task.
+        This includes:
+        - extending the task_config attribute AttrDict to include parameters required for this task
+        - instantiate the Jedi attribute object
+
+        Parameters
+        ----------
+        config: Dict
+            dictionary object containing task configuration
+
+        Returns
+        ----------
+        None
+        """
         super().__init__(config)
 
-        _res_det = int(self.task_config['CASE'][1:])
-        _res_ens = int(self.task_config['CASE_ENS'][1:])
+        _res = int(self.task_config['CASE_ENS'][1:])
+        self.task_config['CASE'] = self.task_config['CASE_ENS']
         _window_begin = add_to_datetime(self.task_config.current_cycle, -to_timedelta(f"{self.task_config['assim_freq']}H") / 2)
-        _recenter_yaml = os.path.join(self.task_config.DATA, f"{self.task_config.RUN}.t{self.task_config['cyc']:02d}z.land_recenter.yaml")
+
+        # fix ocnres
+        self.task_config.OCNRES = f"{self.task_config.OCNRES :03d}"
+
+        # we do not have enkf obs
+        oprefix = self.task_config.RUN.replace("enkf", "")
 
         # Create a local dictionary that is repeatedly used across this class
         local_dict = AttrDict(
             {
-                'npx_ges': _res_ens + 1,
-                'npy_ges': _res_ens + 1,
+                'npx_ges': _res + 1,
+                'npy_ges': _res + 1,
                 'npz_ges': self.task_config.LEVS - 1,
                 'npz': self.task_config.LEVS - 1,
                 'SNOW_WINDOW_BEGIN': _window_begin,
                 'SNOW_WINDOW_LENGTH': f"PT{self.task_config['assim_freq']}H",
-                'ATM_WINDOW_BEGIN': _window_begin,
-                'ATM_WINDOW_LENGTH': f"PT{self.task_config['assim_freq']}H",
-                'OPREFIX': f"{self.task_config.RUN}.t{self.task_config.cyc:02d}z.",
+                'OPREFIX': f"{oprefix}.t{self.task_config.cyc:02d}z.",
                 'APREFIX': f"{self.task_config.RUN}.t{self.task_config.cyc:02d}z.",
-                'jedi_yaml': _recenter_yaml,
+                'GPREFIX': f"gdas.t{self.task_config.previous_cycle.hour:02d}z.",
+                'snow_obsdatain_path': f"{self.task_config.DATA}/obs/",
+                'snow_obsdataout_path': f"{self.task_config.DATA}/diags/",
+                'snow_bkg_path': './bkg/ensmean',
```

Comment:
Same question as above about the background path using `./`
```suggestion
                'snow_obsdatain_path': os.path.join(self.task_config.DATA,'obs'),
                'snow_obsdataout_path': os.path.join(self.task_config.DATA,'diags'),
                'snow_bkg_path': os.path.join('.','bkg','ensmean'),
```